sdk/core/core-sse/src/utils.ts

```diff
@@ -72,7 +74,11 @@ export function ensureAsyncIterable(stream: NodeJS.ReadableStream | ReadableStre
     return {
       cancel: async () => {
         // drain the stream
-        stream.resume();
+        if (typeof (stream as Readable).destroy === "function"){
+          (stream as Readable).destroy();
```

Comment:
For the node case I remember calling `destroy` on the `IncomingMessage` from `http` would actually destroy the entire socket and force you to renegotiate TLS in the case that you wanted to make another server request. This was the performance killer for `HEAD` requests before we fixed it. 

Not sure why it would be causing things to hang in this case though. Maybe it's fine to destroy here since we are canceling? Hmm

Reply:
I assume you're referring to the fix in https://github.com/Azure/azure-sdk-for-js/pull/24356/. I looked into this further and here are a few interesting findings:
- [pooled sockets are unrefed](https://github.com/nodejs/node/blob/9fe0424baa862005327dd0e3318425a211632f92/lib/_http_agent.js#L488) but for some reason they still hold up the app
- [message.socket.end()](https://github.com/nodejs/node/blob/9fe0424baa862005327dd0e3318425a211632f92/lib/internal/streams/writable.js#L799) seems like a more graceful way to implement cancellation here as it sends a FIN packet to the server to make it aware that it doesn't have to bother sending more data. I think we still lose the socket eventually which is not great. The other option is to let the data flow from the server and ignore it but that could have an impact on billing/perf. I am happy to revisit this again if a better design emerges.

Addressed in https://github.com/Azure/azure-sdk-for-js/pull/28111/commits/08b822de97a08ad55d7634e058d70956cab522e2.